src/hotspot/share/opto/compile.hpp

```diff
@@ -829,18 +829,40 @@ class Compile : public Phase {
 
   const CompilationFailureInfo* first_failure_details() const { return _first_failure_details; }
 
```

Comment:
Enabling the stress code in debug builds too could work. I can give it a try, but it will require some  time for more testing.

Reply:
Enabling the new flag in debug builds works if we weaken some existing asserts to allow artificial failures. I'm considering making this stress mode debug-only.

Reply:
I totally support making the stress mode work in debug builds, this is the mode in which we would get most value out of the bailout stress functionality. Support for product builds is secondary in my opinion (although it may add some value, e.g. helping reproduce issues that only surface in product builds), and may be excluded if there are any concerns about e.g. stability.